spec/features/applications/show_spec.rb

```diff
@@ -86,6 +87,46 @@
       
       click_link("#{@pet_2.name}")
       expect(current_path).to eq("/pets/#{@pet_2.id}")
+
+    end
+  end
+
+  # User Story 4
+  describe "Searching for Pets for an Applicatioin" do
+    it "displays an 'Add a Pet' section if the application has not been submitted (status = 'In Progress')" do 
+      visit "/applications/#{@tom.id}"
+      expect(page).to have_content("Status: In Progress")
+      expect(page).to have_content("Add a Pet to this Application")
+
```

Comment:
We may want to break this into two tests with explicit descriptions like "displays an 'Add a Pet' section if the application has not been submitted" and "does not display an 'Add a Pet' section if the application has been submitted, just to make it really clear.